torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -1406,7 +1410,15 @@ Val* hoistProducerIndex(
     std::vector<IterDomain*> loop_domains,
     const std::unordered_map<IterDomain*, Val*> initial_loop_index_map,
     const std::vector<kir::ForLoop*>& loops,
-    Val* index) {
+    Val* index,
+    bool is_overriden_index) {
+  if (is_overriden_index) {
+    // do not hoist overridden index. It is used by
```

Comment:
Is there any failing test caused by this?

Reply:
Yes with this PR. First of all, I think this is what we should do generally, but previously, `index` was always `TensorIndex`, so there was never a definition of `index`. Now there can be a cast expression, so the hoisting logic attempts to hoist, which fails in `getMaybeIndexedIdToHoist`:

```
 TORCH_INTERNAL_ASSERT(
       indexing.indexMap().find(contig_id_it->second) !=
           indexing.indexMap().end(),
```